File: src/main/cpp/commands/command_Outtake.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/command_Outtake.h"

command_Outtake::command_Outtake(subsystem_Intake* intake, subsystem_LED* LED) : m_Intake{intake}, m_LED{LED} {
  // Use addRequirements() here to declare subsystem dependencies.
  AddRequirements({m_Intake});
  AddRequirements({m_LED});
}

// Called when the command is initially scheduled.
void command_Outtake::Initialize() {
  if(m_Intake->getState() == IntakeConstants::hasCube){
    m_Intake->setState(IntakeConstants::cubeOuttaking);
  } else {
    m_Intake->setState(IntakeConstants::coneOuttaking);
  }
  m_Timer.Restart();
}

// Called repeatedly when this Command is scheduled to run
void command_Outtake::Execute() {}

// Returns true when the command should end.
bool command_Outtake::IsFinished() {
  return m_Timer.Get() > 0.5_s;
}

// Called once the command ends or is interrupted.
void command_Outtake::End(bool interrupted) {
  m_Intake->setState(IntakeConstants::noGamepiece);
}

File: src/main/java/frc/robot/commands/command_Outtake.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import edu.wpi.first.wpilibj.Timer;
import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.Constants.IntakeConstants.intakeState;
import frc.robot.subsystems.subsystem_Intake;
import frc.robot.subsystems.subsystem_LEDs;

public class command_Outtake extends CommandBase {
  /** Creates a new command_Outtake. */
  subsystem_Intake m_Intake;
  subsystem_LEDs m_LEDs;
  Timer m_Timer = new Timer();
  public command_Outtake(subsystem_Intake intake, subsystem_LEDs LEDs) {
    m_Intake = intake;
    m_LEDs = LEDs;
    // Use addRequirements() here to declare subsystem dependencies.
    addRequirements(m_Intake);
    addRequirements(m_LEDs);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {
    m_Intake.setCoastMode();
    if (m_Intake.getState() == intakeState.hasCube){
      m_Intake.setState(intakeState.cubeOuttaking);
    } else {
      m_Intake.setState(intakeState.coneOuttaking);
    }
    m_Timer.restart();
  }

  // Called every time the scheduler runs while the command is scheduled.
  @Override
  public void execute() {}

  // Called once the command ends or is interrupted.
  @Override
  public void end(boolean interrupted) {
    m_Intake.setState(intakeState.noGamepiece);
  }

  // Returns true when the command should end.
  @Override
  public boolean isFinished() {
    return m_Timer.get() > 0.5;
  }
}
